palace.pyx

```diff
@@ -1208,6 +1214,118 @@ cdef class SourceGroup:
         """
         self.impl.destroy()
 
+
+cdef class AuxiliaryEffectSlot:
+    """An effect processor.
+
+    It takes the output mix of zero or more sources,
+    applies DSP for the desired effect (as configured
+    by a given `Effect` object), then adds to the output mix.
+
+    This can be used as a context manager that calls `destroy`
+    upon completion of the block, even if an error occurs.
+
+    Parameters
+    ----------
+    context : Context
+        The context from which the auxiliary effect slot is to be created.
+
+    Raises
+    ------
+    RuntimeError
+        If the effect slot can't be created.
+    """
+    cdef alure.AuxiliaryEffectSlot impl
+
+    def __init__(self, context: Context) -> None:
+        self.impl = (<Context> context).impl.create_auxiliary_effect_slot()
+
+    def __enter__(self) -> AuxiliaryEffectSlot:
+        return self
+
+    def __exit__(self, exc_type: Optional[Type[BaseException]],
+                 exc_val: Optional[BaseException],
+                 exc_tb: Optional[TracebackType]) -> Optional[bool]:
+        self.destroy()
+
+    def __lt__(self, other: Any) -> bool:
+        if not isinstance(other, AuxiliaryEffectSlot):
+            return NotImplemented
+        return self.impl < (<AuxiliaryEffectSlot> other).impl
+
+    def __le__(self, other: Any) -> bool:
+        if not isinstance(other, AuxiliaryEffectSlot):
+            return NotImplemented
+        return self.impl <= (<AuxiliaryEffectSlot> other).impl
+
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, AuxiliaryEffectSlot):
+            return NotImplemented
+        return self.impl == (<AuxiliaryEffectSlot> other).impl
+
+    def __ne__(self, other: Any) -> bool:
+        if not isinstance(other, AuxiliaryEffectSlot):
+            return NotImplemented
+        return self.impl != (<AuxiliaryEffectSlot> other).impl
+
+    def __gt__(self, other: Any) -> bool:
+        if not isinstance(other, AuxiliaryEffectSlot):
+            return NotImplemented
+        return self.impl > (<AuxiliaryEffectSlot> other).impl
+
+    def __ge__(self, other: Any) -> bool:
+        if not isinstance(other, AuxiliaryEffectSlot):
+            return NotImplemented
+        return self.impl >= (<AuxiliaryEffectSlot> other).impl
+
+    def __bool__(self) -> bool:
+        return <boolean> self.impl
+
+    def set_gain(self, value: float) -> None:
+        self.impl.set_gain(value)
+
+    def set_send_auto(self, value: bool) -> None:
+        self.impl.set_send_auto(value)
+
+    gain = property(fset=set_gain, doc=(
+        """Set gain for the effect slot.
+        """))
```

Comment:
Please make this a one-liner, i.e. `gain = property(fset=set_gain, doc='Gain of the effect slot.')`.  Remember that it's a property, so it should be documented as an object, not a function.

Reply:
As for the other two properties, should I keep it as is, or make it single-quote multi-line?

E.g.
```
'If set to `True`, the reverb effect will automatically'
'apply adjustments to the source\'s send slot gains based'
...
```

Reply:
I'd prefer multi-line docstrings to be triple quoted (less noisy to my eyes).